### ISSUE TITLE
Make a new release 0.9.0, RSN

### ISSUE THREAD

**Issue by deeplook** [deeplook/svglib#140]
There have been many requests for a new release of svglib on PyPI, which I have procrastinated for way too long (my sincerest apologies!). I've collected some thoughts in https://github.com/deeplook/svglib/wiki/Release-Process, but in the end it will likely happen rather by mimicking a few techniques I found to be successfully used when instantiating projects with https://github.com/audreyr/cookiecutter/ using a template like https://github.com/audreyr/cookiecutter-pypackage which I've used for something different, before. It relies on a couple of tools like `bumpversion` (not really maintained anymore) and `twine`, but using anything else will be better than what there is now (not much).

**Comment by claudep** [comment]
I'm happily using `twine` for some of my projects.

**Comment by claudep** [comment]
I wonder if it would be worth shipping #116 (CSS support) in that release, even if it's not polished (better than no support at all?).

**Comment by deeplook** [comment]
I would put in everything new since 0.8.1, as simple as that.

**Comment by claudep** [comment]
Well, PR #116 is not merged yet, that's why I'm asking.

**Comment by deeplook** [comment]
I won't have any time to test this in any other way than reviewing code and test cases, and build results from travis-ci. Just looked briefly now. Will comment in https://github.com/deeplook/svglib/issues/116.

**Comment by MinchinWeb** [comment]
I would add a request for a new PyPI version. There's a small error in version 0.8.1 (and 0.8.0) where `xrange` [is called](https://github.com/deeplook/svglib/blob/v0.8.1/svglib/svglib.py#L747), but that function is not available on Python 3. So right now, Python 3 compatibility is broken. But the this has been fixed on GitHub.

Thanks for the library!

**Comment by Overdrivr** [comment]
Would you like some help for automating the release of the library ?
Something I rely extensively on is configuring appveyor/travis to deploy only on tags.
This way, the only thing you need to perform a release is to create a new tag in Github.

**Comment by deeplook** [comment]
@Overdrivr We have appveyor and travis set up for svglib. As you offer help that might accelerate things. I would like to keep appveyor and travis checking every commit, but if we can have releases with every new tag that would be fine. If you have more info to point me to I will be glad to have a look!

**Comment by Overdrivr** [comment]
Both things are not exclusive, on the contrary. Basically the idea is to run the same build & test suite, whether it's a simple commit or a commit with tag. And simply on tags, go the extra step and push to pypi. 

I'll open a PR then, it will give you a better idea.

**Comment by deeplook** [comment]
https://pypi.org/project/svglib/0.9.0b0/ has been just published, although it's not automatically visible when going to https://pypi.org/project/svglib/, most likely because of the beta0 flag.

**Comment by deeplook** [comment]
Let's have two weeks of testing and clean-up and then make a final release 0.9.0!

**Comment by deeplook** [comment]
I was planning to release 0.9.0, without beta in 1-2 days. Not much has happened during the last two weeks, but I'll do it nevertheless, because then this version gets more visibility on PyPI, and might trigger some more motivation to clean-up stuff and go for 1.0.

Maybe @nicoddemus could provide some insight about how to create conda packages, too, if I promise to buy his [new book](https://www.packtpub.com/web-development/pytest-quick-start-guide)? :)

**Comment by claudep** [comment]
That's why I prefer releasing plain .0 packages, then release .1 quickly if bugs arise. beta makes sense when you really know that people will make the extra effort to test those versions, which is unfortunately pretty rare (and which they could also do by pip-installing from Github directly).

**Comment by nicoddemus** [comment]
> Maybe @nicoddemus could provide some insight about how to create conda packages, too, if I promise to buy his new book? :)

Hehehe gladly. 😁

**Comment by nicoddemus** [comment]
By the way, there's already a [conda-recipe in conda-forge](https://github.com/conda-forge/svglib-feedstock), so as soon as this is released to PyPI the conda-forge bot will automatically update the recipe. 😁

**Comment by jdgwf** [comment]
It's me again! If you add 

```
try:
    # Python 2
    xrange
except NameError:
    # Python 3, xrange is now named range
    xrange = range
```
Just below the imports 

it works perfectly again in python3.6 :)

**Comment by jdgwf** [comment]
( too small of a  change to make a PR, I will if you need me to )

**Comment by deeplook** [comment]
@jdgwf No need for a PR because of https://github.com/deeplook/svglib/issues/163.

**Comment by jdgwf** [comment]
My apologies; I'm a lazy contributor. I've been swamped this year and just made a quick fix for a personal project. I'll be better, I promise! 💚

**Comment by nicoddemus** [comment]
@deeplook you might consider also using [setuptools-scm](https://github.com/pypa/setuptools_scm) and Travis for deployment. I use it in a number (almost all) my open source projects and works like a charm. 👍

**Comment by deeplook** [comment]
@nicoddemus Thanks, will have a look. But before that I'll push a real 0.9.0, RSN...

**Comment by msom** [comment]
What is the current state? Can you make the release 0.9.0?

**Comment by deeplook** [comment]
I guess I will simply make that release without any further ado within 1-2 days. I’ve learned that making the beta0 was a bad idea, but I also didn’t know that PyPI would sort of hide beta releases.

In the meantime I’ve also learned to continuously make more or less sloppy mini commits and silly releases (fighting with a README I wanted to be in markdown format) by the dozens from browsers, and mobile devices on the go. The history is a mess, but it worked. So, yes, I can make releases. You can see this new emerging pet project here: https://github.com/deeplook/ipyrest.

**Comment by MinchinWeb** [comment]
My two bits: make it easy (on you) to make releases (ideally single command to build, package, and ship), and then release early and often. If you hue to SemVer, then upgrading on the user end isn't a huge pain either.

**Comment by msom** [comment]
Any news on that?

**Comment by deeplook** [comment]
I've just manually published 0.9.0. Again, sorry for the way to long delay! Life is not always going as expected.
